assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -237,21 +237,45 @@ class AssistedInjectProcessor : AbstractProcessor() {
     val factoryExecutable = types.asMemberOf(factoryType.asType() as DeclaredType,
         factoryMethod) as ExecutableType
 
-    val expectedKeys = assistedRequests.map { it.namedKey }.sorted()
+    val expectedKeys = assistedRequests.map { it.namedKey }
     val factoryKeys = factoryMethod.parameters
         .zip(factoryExecutable.parameterTypes) { element, mirror -> element.asNamedKey(mirror) }
-    val keys = factoryKeys.sorted()
-    if (keys != expectedKeys) {
+
+    // Rename any single keys to the factory keys to avoid requiring same names
+    val singleFactoryKeys = factoryKeys
+        .filter { key -> factoryKeys.count { it.key == key.key } == 1 }
+        .toMutableSet()
+    val renameableKeys = expectedKeys
+        .map { expectedKey ->
+          val match = singleFactoryKeys.firstOrNull { factoryKey -> factoryKey.key == expectedKey.key }
+          if (match != null) {
+            singleFactoryKeys.remove(match)
```

Comment:
Could you mutate data from outside the `map` ?
`singleFactoryKeys - expectedKeys` should be enough?

Reply:
Mmm, I think this was required during the processing of `expectedKeys` in the following example case:

    class Test {
      @AssistedInject
      Test(@Assisted String fao, @Assisted String faz, String bar) {}

      @AssistedInject.Factory
      interface Factory {
        Test create(String foo);
      }
    }

I'll rewrite this so that it more cleanly only matches single requests against single factory keys.